modules/ROOT/pages/indexes/semantic-indexes/vector-indexes.adoc

```diff
@@ -117,12 +119,58 @@ You can also create a vector index for relationships with a particular type on a
 ----
 CREATE VECTOR INDEX name IF NOT EXISTS
 FOR ()-[r:REL_TYPE]-() ON (r.embedding)
-OPTIONS {indexConfig: {
+OPTIONS { indexConfig: {
  `vector.dimensions`: $dimension,
  `vector.similarity_function`: $similarityFunction
 }}
 ----
 
+[[configuration-settings]]
+=== Configuration settings
+
+For differences with accepted values between vector index providers, see xref:indexes/semantic-indexes/vector-indexes.adoc#vector-index-providers[].
+
+`vector.dimensions` ::
+The dimensionality of the vectors to be indexed, see xref:indexes/semantic-indexes/vector-indexes.adoc#embeddings[].
+Omitting this setting is possible, and any `LIST<INTEGER | FLOAT>` can be indexed and queried, separated by their dimensionality.
+Setting this value adds additional checks that ensure only vectors with the configured dimensionality are indexed, and querying the index with a vector of a different dimensionality returns an error. It is recommended to provide the dimensions.
+Accepted values::: `INTEGER` between `1` and `4096` inclusively.
+Default value::: None, though the setting can be omitted. The setting was mandatory prior to Neo4j 5.23.
+
+`vector.similarity_function` ::
+The name of the similarity function to use to assess how similar two vectors are.
+To read more about the available similarity functions, see xref:indexes/semantic-indexes/vector-indexes.adoc#similarity-functions[].
+
+Accepted values::: `STRING`: `'cosine'`, `'euclidean'`.
+Default value::: `'cosine'`. The setting was mandatory prior to Neo4j 5.23.
+
+`vector.quantization.enabled` label:new[Introduced in 5.23] ::
+Quantization is a technique to reduce the size of vector representations.
+Enabling quantization can accelerate search performance but can slightly decrease accuracy.
+This is recommended to always be enabled on machines with limited memory.
+Vector indexes created prior to Neo4j 5.23 have this setting effectively set as `false`.
+
+Accepted values::: `BOOLEAN`: `true`, `false`.
+Default value::: `true`
+
+==== Advanced configuration settings
+
+`vector.hnsw.m` label:new[Introduced in 5.23] ::
+The `M` parameter controls the maximum number of connections each node has in the HNSW graph.
+Increasing this value may lead to greater accuracy at the expense of increased index population and update times, especially for vectors with high dimensionality.
+Vector indexes created prior to Neo4j 5.23 have this setting effectively set as `16`.
+
+Accepted values::: `INTEGER` between `1` and `512` inclusively.
+Default value::: `16`
+
+`vector.hnsw.ef_construction` label:new[Introduced in 5.23] ::
+The number of nearest neighbors tracked during the insertion of vectors into the HNSW graph.
+Increasing this value increases the quality of the index, and may lead to greater accuracy (with diminishing returns) at the expense of increased index population and update times.
```

Comment:
It's quite a normal phrase. Though I was struggling where to exactly put it in the sentence.

Basically, increasing `efConstruction` makes a larger change to accuracy initially; however, that perceived change diminishes as you keep increasing it; especially as it gets more expensive as you increase it. The value to time ratio decreases.

Pulling random numbers:
| input | output |
| - | - |
| 1 | 10.00 |
| 2 | 20.00 |
| 3 | 25.00 |
| 4 | 27.50 |
| 5 | 28.75 |

This sequence has diminishing returns.

A few other venders note this, and also use "diminishing returns"
I believe it is important to note something like this, ideally with that or similar phrasing; otherwise a user may just configure it to the max and then get something extremely expensive, and very little difference in accuracy compared to a much smaller value.

Again, these are "advanced" settings.

Reply:
Ok. I know the meaning of the phrase, but was unsure about its precise meaning in this context, so thank you for explaining. I think it is probably fine as it stands 👍

Reply:
Sorry, I wasn't sure 😅